rootfs_test.go

```diff
@@ -346,3 +382,89 @@ func TestPartitionsAPI(t *testing.T) {
 		mt.AssertTrue(t, err != nil)
 	}
 }
+
+func TestNetworkRootfs(t *testing.T) {
+	prepareMockDevices(t)
+	defer cleanupMockDevices()
+
+	var server http.Server
+
+	server.Handler = http.FileServer(http.Dir("."))
+	addr := ":8081"
+	listen, err := net.Listen("tcp", addr)
+	mt.AssertNoError(t, err)
+
+	defer listen.Close()
+	go server.Serve(listen)
+
+	// Do this test twice, once with a valid update, and once with a too
+	// short/broken update.
+	for _, mode := range []int{0, os.O_TRUNC}[:] {
```

Comment:
Maybe just for readability we can put this below in separate function like `ExecuteNetworkTest` or similar.